src/spatialdata/_core/spatialdata.py

```diff
@@ -492,93 +496,83 @@ def aggregate(
             **kwargs,
         )
 
-    @staticmethod
-    def _validate_unique_element_names(element_names: list[str]) -> None:
-        if len(element_names) != len(set(element_names)):
-            duplicates = {x for x in element_names if element_names.count(x) > 1}
-            raise KeyError(
-                f"Element names must be unique. The following element names are used multiple times: {duplicates}"
-            )
-
     def is_backed(self) -> bool:
         """Check if the data is backed by a Zarr storage or if it is in-memory."""
-        return self._path is not None
+        return self.path is not None
 
     @property
     def path(self) -> Path | None:
         """Path to the Zarr storage."""
-        return self._path
+        if self._path is None:
+            return None
+        if isinstance(self._path, str):
+            return Path(self._path)
+        if isinstance(self._path, Path):
+            return self._path
```

Comment:
I would prefer that we do the type coercion on the setter. That is, `SpatialData._path` should always be a `Path` object. That way the internal representation is consistent.

Reply:
Good point! Done.